src/daemon/plugin.h

```diff
@@ -243,21 +243,19 @@ int plugin_shutdown_all(void);
  *  plugin_write
  *
  * DESCRIPTION
- *  Calls the write function of the given plugin with the provided data set and
- *  value list. It differs from `plugin_dispatch_values' in that it does not
- *  update the cache, does not do threshold checking, call the chain subsystem
- *  and so on. It looks up the requested plugin and invokes the function, end
- *  of story.
+ *  Directly enqueues the given metric family without updating the cache or
+ *  calling the chain subsystem. The metric family may still be dropped from
+ *  the queue if collectd is stopped or the queue length surpasses the
+ *  configured limit.
  *
  * ARGUMENTS
  *  plugin     Name of the plugin. If NULL, the value is sent to all registered
  *             write functions.
- *  m   The actual value to be processed. Must not be NULL.
+ *  fam   The actual metric family to be processed.
```

Comment:
It still must not be NULL, right?

Reply:
The first thing `plugin_write` does is to check if `fam` is `NULL`, so nothing bad would happen if it was `NULL`.
I guess that's why I removed that part of the documentation.

The behavior between the previous `plugin_write` and the new `plugin_write` with regards to `fam` being `NULL` hasn't actually changed, so it makes sense to keep the documentation the same.
I've added a commit that re-adds the `Must not be NULL`.

Reply:
This can be set as resolved now too.